uber/models.py

```diff
@@ -1523,16 +1543,44 @@ def weighted_hours(self):
     def total_hours(self):
         return self.weighted_hours * self.slots
 
-    @cached_property
-    def all_staffers(self):
-        return self.session.query(Attendee).order_by(Attendee.last_first).all()
+    def _potential_staff(self, staff_only=False, order_by=Attendee.full_name):
+        """
+        return a list of attendees who:
+        1) are assigned to this job's location
+        2) are allowed to work this job (job is unrestricted, or they're trusted in this job's location)
 
-    @cached_property
-    def available_staffers(self):
-        return [s for s in self.all_staffers
-                if self.location in s.assigned_depts_ints
-                   and (s.trusted or not self.restricted)
-                   and self.no_overlap(s)]
+        :param: staff_only: restrict result to attendees where staffing==True
+        :param: order_by: order by another Attendee attribute
+        """
+        return (self.session.query(Attendee)
+                .filter(Attendee.assigned_depts.contains(str(self.location)))
+                .filter(*[Attendee.trusted_depts.contains(str(self.location))] if self.restricted else [])
+                .filter_by(**{'staffing': True} if staff_only else {})
+                .order_by(order_by)
+                .all())
+
+    @property
+    def capable_staff_opts(self):
+        # format output for use with the {% options %} template decorator
+        return [(a.id, a.full_name) for a in self.capable_staff]
+
+    @property
+    def capable_staff(self):
+        """
+        Return a list of staffers who could sign up for this job.
+
+        Important: Just because a staffer is capable of working
+        this job doesn't mean they are available to work.
+        They may have other shift hours during that time period.
+        """
+        return self._potential_staff(staff_only=True)
+
+    @property
+    def available_staff(self):
+        """
+        Return a list of staffers who are allowed to sign up for this Job and have the free time
+        """
+        return [s for s in self._potential_staff(order_by=Attendee.last_first) if self.no_overlap(s)]
 
```

Comment:
It all looks neat and clean right now but it took me like 30 minutes to sit there and really think about what these should be named. As Eli says, there are 2 hard problems in computer science: caching, and naming things.  This refactor had both : )